SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -589,24 +622,27 @@ void UGlobalStateManager::QueryGSM(bool bRetryUntilAcceptingPlayers)
 		}
 		else
 		{
-			bool bNewAcceptingPlayers = GetAcceptingPlayersFromQueryResponse(Op);
+			bool bNewAcceptingPlayers = false;
+			int32 QuerySessionId = 0;
+			bool bQueryResponseSuccess = GetAcceptingPlayersAndSessionIdFromQueryResponse(Op, bNewAcceptingPlayers, QuerySessionId);
```

Comment:
I still think that `bQueryResponseSuccess` should be checked in a different if statement so we can log a separate error for it. We can still retry the query but want to make the user aware something could be wrong.
If `bQueryResponseSuccess = false` the GSM is potentially in a broken state and the deployment will never recover. 

For `bNewAcceptingPlayers != bExpectedAcceptingPlayers || QuerySessionId != ExpectedSessionId` we are just waiting for an expected state.